test/AutoRest.Shared.Tests/WriterExtensionTests.cs

```diff
@@ -87,6 +103,17 @@ public static void WriteObjectValueNullIUtf8JsonSerializable ()
             Assert.AreEqual("null", System.Text.Encoding.UTF8.GetString(stream.ToArray()));
         }
 
+        [Test, TestCaseSource("BinaryDataCases")]
+        public static void WriteBinaryDataValue(BinaryData value, string expectedValue)
+        {
+            using MemoryStream stream = new MemoryStream();
+            using Utf8JsonWriter writer = new Utf8JsonWriter(stream);
+            writer.WriteBinaryDataValue(value);
+
+            writer.Flush();
+            Assert.AreEqual(expectedValue, System.Text.Encoding.UTF8.GetString(stream.ToArray()));
```

Comment:
Can we add round trip tests to serialize, then deserialize and validate the original byte array is equal to the new byte array.

I don't know if it will make a difference but can we also validate these cases while they are a property of a larger object?

Reply:
I have added a test case in MgmtExtensionResourceTests to see if the result is expected when we first serialize the model, which contains a binary data property and then deserialize it.